### PR TITLE
185 refactor onboarding remove user button

### PR DESCRIPTION
**Pull request by LeanderCretegny**
This PR removes user button in onboarding, as well as redundant back button, and fix some tests to fit the UI changes.

Testing: go through the onboarding and/or the settings to check that the views are not displaying the user button and back button anymore.

### CODE REVIEW

File: lib/widgets/wizard_widgets.dart [hunk 1]
```diff
@@ -177,7 +177,12 @@ class WizardScaffold extends StatelessWidget {
   final PreferredSizeWidget topBar;
   final Widget? body;
 
-  const WizardScaffold({this.topBar = const TopAppBar(), this.body, super.key});
+  const WizardScaffold(
+      {this.topBar = const TopAppBar(
+        enableProfileButton: false,
+      ),
+      this.body = const Text("unimplemented"),
```
Comment (Thechi2000) [inline lib/widgets/wizard_widgets.dart:184]: Should use the localized version (as per #174)

File: test/integration/onboarding.dart [hunk 2]
```diff
@@ -125,6 +125,9 @@ void main() async {
     await tester.tap(find.textContaining("Next"));
     await tester.pumpAndSettle();
 
+    await tester.tap(find.text("Skip"));
```
Comment (Thechi2000) [inline test/integration/onboarding.dart:128]: Should use the localized version (as per https://github.com/ActualIA/app/pull/174)

**Review by Thechi2000 — CHANGES_REQUESTED** [review]
Small and efficient, thanks ! Although it is missing the internationalization.

File: lib/widgets/wizard_widgets.dart (hunk 1 above)
Comment (LeanderCretegny) [reply lib/widgets/wizard_widgets.dart:184]: I don't see why, there is no displayed text here (the text saying "unimplemented" is for dev only, as it is a placeholder)

File: test/integration/onboarding.dart (hunk 2 above)
Comment (LeanderCretegny) [reply test/integration/onboarding.dart:128]: Is it really necessary inside a test?

Comment (Thechi2000) [reply test/integration/onboarding.dart:128]: Oh my bad, I didn't see it was in a test 😅

File: lib/widgets/wizard_widgets.dart (hunk 1 above)
Comment (Thechi2000) [reply lib/widgets/wizard_widgets.dart:184]: We did it until then, I think it would be good for consistency

**Review by quartztz — APPROVED** [review]
Much needed! Now it's consistent.

Thanks for localizing as well ! LGTM as is

**Review by quartztz — APPROVED** [review]
Much needed! Now it's consistent with expectations (and internationalized thanks to @Thechi2000's pedantry :) ). LGTM! Can be merged as is.

**Review by quartztz — DISMISSED** [review]
Much needed! Now it's consistent with expectations (and internationalized thanks to @Thechi2000's pedantry :) ). LGTM! Can be merged as is.

File: lib/views/providers_wizard_view.dart [hunk 3]
```diff
@@ -71,8 +71,8 @@ class _ProvidersWizardView extends State<ProvidersWizardView> {
           rOnPressed: () async {
             pvm.updateProvidersFromEdited();
             if (!await pvm.pushNewsProviders(loc)) {
-              if (context.mounted && Platform.isAndroid) {
-                Fluttertoast.showToast(msg: loc.providersUpdateError);
+              if (Platform.isAndroid) {
+                Fluttertoast.showToast(msg: "Error while updating providers");
```
Comment (Thechi2000) [inline lib/views/providers_wizard_view.dart:75]: Nice catch for this nasty bug :)

However, you should not get rid of the localization of the message.

**Review by Thechi2000 — CHANGES_REQUESTED** [review]
One last little thing, then it will be ready to merge imo

**Review by Pacsort17 — APPROVED** [review]
Small bug detected, but not related to this PR (I will open an issue)

Tested on Android, buttons are indeed removed.

**Review by Thechi2000 — APPROVED** [review]

**Review by Pacsort17 — APPROVED** [review]

**Review by Thechi2000 — APPROVED** [review]

**Pull request merged**